ellipticcurve/private_key.go

```diff
@@ -0,0 +1,102 @@
+package ellipticcurve
+
+import (
+	"crypto/rand"
+	"math/big"
+	"starkbank/ecdsa-go/ellipticcurve/utils"
+)
+
+type PrivateKey struct {
+	Curve  CurveFp
+	Secret *big.Int
+}
+
+func NewPrivateKey(curve CurveFp, secret *big.Int) PrivateKey {
+	privateKey := PrivateKey{
+		Curve:  curve,
+		Secret: secret,
+	}
+	randomSecret, err := rand.Int(rand.Reader, curve.N)
```

Comment:
Pode reutilizar o **secret** que já existe se não for informado

Reply:
Também pode colocar antes ou tentar colocar inline pra ficar igual ao starkbank/ecdsa-python